test/merge-with-rules.test.ts

```diff
@@ -1187,6 +1259,83 @@ describe("Merge with rules", function () {
         },
       })(conf1, conf2)
     ).toEqual(expected);
-  })
+  });
+
+  it("should merge with same with different options - 2 #169", () => {
```

Comment:
In this case, as opposed to the previous one, `exclude` has explicit merge rule, and everything else matches, so it seems legit to me.